packages/peregrine/lib/talons/CartPage/ProductListing/useProduct.js

```diff
@@ -48,21 +52,23 @@ export const useProduct = props => {
 
     const [isFavorite, setIsFavorite] = useState(false);
 
-    const updateItemErrorMessage = useMemo(() => {
-        if (!updateError) return null;
+    const derivedErrorMessage = useMemo(() => {
+        if (!updateError && !removeItemError) return null;
 
-        if (updateError.graphQLErrors) {
+        const errorTarget = updateError || removeItemError;
+
+        if (errorTarget.graphQLErrors) {
             // Apollo prepends "GraphQL Error:" onto the message,
             // which we don't want to show to an end user.
             // Build up the error message manually without the prepended text.
-            return updateError.graphQLErrors
+            return errorTarget.graphQLErrors
                 .map(({ message }) => message)
                 .join(', ');
         }
 
         // A non-GraphQL error occurred.
-        return updateError.message;
-    }, [updateError]);
+        return errorTarget.message;
```

Comment:
I kept this pattern around where I was re-factoring and it existed, but noticed on more recent code we don't use it anymore. Can this be removed, and we only need to care about GraphQL errors?

Reply:
How would we get here, network error?

Reply:
After further review, you are correct! This is needed.